### DISCUSSION TITLE
PVP CLAIM

### DISCUSSION THREAD

**Discussion by giantfurkii** [GriefPrevention/GriefPrevention#2254] [Ideas]
When 2 players are in battle, another player comes and claims the location of the players who are in the battle.

**Comment by bigpresh** [comment]
That's something of a rare edge case, I'd say.  I'm not personally sure it's worth the effort to add a check that there are no players within the claim area engaged in PvP... and if we did, you could then argue that it allows people to prevent other players from claiming.

It could quite probably be implemented as an addon by someone who wanted to, though - listening for the `ClaimCreatedEvent` event, finding players within the claimed area (maybe `getNearbyEntities​()` with a `BoundingBox` constructed from the new claim's coords) and checking if they're involved in PvP (Get the player's data from GP with `getPlayerData()`, check `playerData.inPvpCombat()`) and cancelling the event (and sending an explanatory message) if so.
